### DISCUSSION TITLE
400 status code/  getUserUid axios Call

### DISCUSSION THREAD

**Discussion by fimoefive** [nss-evening-cohort-14/evening-client-side#327] [Capstones!] [answered]
Martin

**issue:**
Trying to resolve this 400 status bad request error code before pushing up this current working branch.

The user uid authentication works with specific users, but I am getting a 400 status code for the getUserUid axios.get call in my console.
I'm not sure if its my Firebase Rules or useEffect for Firebase in the App() function.

xhr.js:177 GET https://golf-app-962e7-default-rtdb.firebaseio.com/user.json?orderBy=%22uid%22&equalTo=%22undefined%22 400 (Bad Request)

createError.js:16 Uncaught (in promise) Error: Request failed with status code 400
    at createError (createError.js:16)
    at settle (settle.js:17)
    at XMLHttpRequest.handleLoad (xhr.js:62)

**What I have tried:**

const getUserUid = (user) => new Promise((resolve, reject) => {
  axios.get(`${dbUrl}/user.json?orderBy="uid"&equalTo="${user.uid}"`)
    .then((response) => resolve(response))
    .catch((error) => reject(error));
});

**I've also setup my Firebase rules to:**
            "games": {
        ".indexOn": ["uid", "gameFirebaseKey"]
      }, 
            "holes": {
        ".indexOn": ["uid","hole1", "hole2", "hole3", "hole4", "hole5", "hole6", "hole7", "hole8", "hole9", ".value"]
      }, 

**Google links:**
https://stackoverflow.com/questions/43260689/adding-and-indexon-value-into-firebase-rules

Links to code:

https://github.com/fimoefive/Golf-ScoreCard-App/blob/mjs-uid/src/App/App.js

**Comment by Gabrielle-Tobermann** [comment]
Your axios call is ```axios.get(${dbUrl}/user.json?orderBy="uid"&equalTo="${user.uid}")``` do you have only one user? Or is it supposed to be ```users.json``` ?

**Reply by fimoefive** [reply]
I have two users, but one with a fake uid string in my user table in firebase.
Everything is working like it should, but I don't want to push up this working branch until this 400 error is resolved.

**Reply by drteresavasquez** [reply]
@fimoefive give us a screen shot of your nodes in firebase. @Gabrielle-Tobermann is asking if your route should be plural, which is exactly what I was about to ask. What does a 400 error mean?

**Reply by drteresavasquez** [reply]
Also, your error is telling you that the UID you are passing is undefined: ` https://golf-app-962e7-default-rtdb.firebaseio.com/user.json?orderBy=%22uid%22&equalTo=%22undefined%22 `

Make sure that you are actually passing the right thing to the function

**Comment by drteresavasquez** [comment]
@fimoefive will update with resolution

**Comment by fimoefive** [comment, marked as answer]
I Didn't have my Firebase Rules for user table published:    
               "user": {
        ".indexOn": ["uid"]
      }, 
  }
  
  Also it is Important and Dr. T brought to my attention that its imperative to import "import firebase from 'firebase/app'; import 'firebase/auth';" and intitalize Firebase from index.js file than in the App() function. 
 This prevents having to initialize within the App() every time the URL changes. Also make the App run faster.
